hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java

```diff
@@ -217,6 +217,11 @@ protected boolean shouldKillAfterStoreUpdate(final boolean isSuspended) {
    */
   private TimeoutExecutorThread<TEnvironment> timeoutExecutor;
 
+  /**
+   * Use a dedicated thread for executing WorkerMonitor.
```

Comment:
Add more comments here to explain why we need a dedicated thread?

Reply:
Ok, done.